src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -41,18 +48,24 @@ public class ResourceDependencyVisitor : SyntaxVisitor
                     : @group)
                 .ToImmutableHashSet();
 
-        private ResourceDependencyVisitor(SemanticModel model)
+        private ResourceDependencyVisitor(SemanticModel model, bool faultTolerant)
         {
             this.model = model;
+            this.faultTolerant = faultTolerant;
             this.resourceDependencies = new Dictionary<DeclaredSymbol, HashSet<ResourceDependency>>();
             this.currentDeclaration = null;
         }
 
         public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax syntax)
         {
-            if (!(this.model.GetSymbolInfo(syntax) is ResourceSymbol resourceSymbol))
+            if (this.model.GetSymbolInfo(syntax) is not ResourceSymbol resourceSymbol)
             {
-                throw new InvalidOperationException("Unbound declaration");
+                if (!this.faultTolerant)
+                {
+                    throw new InvalidOperationException("Unbound declaration");
```

Comment:
>InvalidOperationException [](http://example.com/codeflow?start=30&length=25)

The reason it was throwing originally is because it was written for codegen which runs only if there are no errors and we didn't have a need to handle these types of issues gracefully.

Now we have a need for it so we could just permanently make it fault tolerant to simplify the logic.

Reply:
Makes sense. Will simplify it.